From: Jiri Slaby <jslaby@suse.cz>
Date: Tue, 14 Mar 2017 09:08:47 +0100
Subject: Revert "RDMA/core: Fix incorrect structure packing for booleans"
Patch-mainline: never, kabi
References: kabi

This reverts commit 600bfb438cd9ab51f952bc44364c20e82d184fa4, upstream
commit 55efcfcd7776165b294f8b5cd6e05ca00ec89b7c. It breaks kABI and as
far as I can see the structure is used over drivers. Given this is a bug
since ever and we have not hit it yet, I assume we can just revert it
for now.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/rdma/ib_sa.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/include/rdma/ib_sa.h
+++ b/include/rdma/ib_sa.h
@@ -137,12 +137,12 @@ struct ib_sa_path_rec {
 	union ib_gid sgid;
 	__be16       dlid;
 	__be16       slid;
-	u8           raw_traffic;
+	int          raw_traffic;
 	/* reserved */
 	__be32       flow_label;
 	u8           hop_limit;
 	u8           traffic_class;
-	u8           reversible;
+	int          reversible;
 	u8           numb_path;
 	__be16       pkey;
 	__be16       qos_class;
@@ -193,7 +193,7 @@ struct ib_sa_mcmember_rec {
 	u8           hop_limit;
 	u8           scope;
 	u8           join_state;
-	u8           proxy_join;
+	int          proxy_join;
 };
 
 /* Service Record Component Mask Sec 15.2.5.14 Ver 1.1	*/
